File: subsidy-platform/src/main/java/com/subsidy_platform/service/ApplicationService.java
package com.subsidy_platform.service;

import com.subsidy_platform.dto.ApplicationRequest;
import com.subsidy_platform.dto.ApplicationResponse;
import com.subsidy_platform.dto.VerificationRequest;

import java.util.List;

public interface ApplicationService {
    List<ApplicationResponse> getAll();
    List<ApplicationResponse> getForRole(String role, String email);
    ApplicationResponse getById(Long id, String role, String email);
    ApplicationResponse create(ApplicationRequest request, String role, String email);
    ApplicationResponse verify(Long id, String role, VerificationRequest request);
    ApplicationResponse reverify(Long id, String role, String remarks);
}

File: frontend/src/services/applicationService.ts
import apiClient from './apiClient';
import type {
  ApplicationRequest,
  ApplicationResponse,
  VerificationRequest,
} from '../types/api';

export const applicationService = {
  getAll: async (): Promise<ApplicationResponse[]> => {
    const response = await apiClient.get<ApplicationResponse[]>('/applications');
    return response.data;
  },

  getPending: async (): Promise<ApplicationResponse[]> => {
    const response = await apiClient.get<ApplicationResponse[]>('/applications/pending');
    return response.data;
  },

  getById: async (id: number): Promise<ApplicationResponse> => {
    const response = await apiClient.get<ApplicationResponse>(`/applications/${id}`);
    return response.data;
  },

  create: async (data: ApplicationRequest): Promise<ApplicationResponse> => {
    const response = await apiClient.post<ApplicationResponse>('/applications', data);
    return response.data;
  },

  verify: async (id: number, data: VerificationRequest): Promise<ApplicationResponse> => {
    const response = await apiClient.put<ApplicationResponse>(`/applications/${id}/verify`, data);
    return response.data;
  },

  reverify: async (id: number, remarks?: string): Promise<ApplicationResponse> => {
    const response = await apiClient.put<ApplicationResponse>(
      `/applications/${id}/reverify`,
      remarks ? { remarks } : {}
    );
    return response.data;
  },
};

export default applicationService;
